### ISSUE TITLE
FEATURE: Change HOST email FREQUENCY based on IMMEDIACY (date/time related to current date/time)

### ISSUE THREAD

**Issue by NanKrafft** [DevProgress/HillaryBNB#57]
Emails to hosts listing inquiries that day continue to go out nightly (are dated), BUT if a traveler user selects a start date that is within 48 hours of the current time/date the email to the host goes out immediately (date and time stamp)

If possible > ADD "Urgent" to subject or within email 

Email copy here: 
https://docs.google.com/document/d/1cxThVacGKHSMLKl6MX5CpEasXvj613l2f9eOo8MLXJQ/edit?ts=57fbfeff

ADD: Comms for hosts to FAQ
-  encourage hosts to update their listing and remove it once space is filled.

FAQ copy here: 
https://docs.google.com/document/d/1U6KCFfYO6z3KeXa9D2bYc_XnQ_xM2gmKLUXE5VV8H5M/edit#

**Comment by dsjoerg** [comment]
@NanKrafft should we go ahead and implement this or do you want to work on it/think on it more?

**Comment by NanKrafft** [comment]
@dsjoerg I think we can move on it. Will need to sync with dev on updates to copy, but we have time for that. :)

**Comment by dsjoerg** [comment]
@NanKrafft to be sure I understand what we want here, see [this](https://github.com/devprogress/hillarybnb#when-does-hillarybnb-send-emails) for current email-sending logic.

New proposed logic:
- When a visitor clicks "SEND MY CONTACT INFO", if their visit will begin within 48 hours, the email is sent immediately to the host (an "Instant Email").
- Every night at 3:30/4am Eastern time we do the following:
- For each Hosting Offer registered in the system, if there are any visitors who clicked the "SEND MY CONTACT INFO" button within the past 24 hours (and for which we did not send an Instant Email), we gather their contact information and email them to the the host. (Note that this means a host may receive multiple emails from us if they have multiple Hosting Offers).
- For each Visit registered in the system, if there are any new Hosting Offers created within the past 24 hours that are within 20 miles of the Visit's zip code, we email them to the visitor. (Note that this means a visitor may receive multiple emails from us if they have multiple pending Visits).

Is that right?

**Comment by NanKrafft** [comment]
@dsjoerg 

**YES** - When a visitor clicks "SEND MY CONTACT INFO", if their visit will begin within 48 hours, the email is sent immediately to the host (an "Instant Email").

Every night at 3:30/4am Eastern time we do the following:

**YES** - For each Hosting Offer registered in the system, if there are any visitors who clicked the "SEND MY CONTACT INFO" button within the past 24 hours (and for which we did not send an Instant Email), we gather their contact information and email them to the the host. (Note that this means a host may receive multiple emails from us if they have multiple Hosting Offers). **They will get multiple emails if... they have 1 or more requests in the next 48hrs, OR 1 or more requests in the next 48hrs AND a request for > 48hrs later.**

^ Think this is a good start and that we can watch and modify if we see an issue. 

For each Visit registered in the system, if there are any new Hosting Offers created within the past 24 hours that are within 20 miles of the Visit's zip code, we email them to the visitor. (Note that this means a visitor may receive multiple emails from us if they have multiple pending Visits). - **My suggestion here would be that IF a host becomes available during any of their visits -> we send a one a day notification that links them back to the page that shows available hosts. Something like "There are now hosts available during one of your visits"** 

**We should also add language that encourages visitors to remove their visits once they have secured accommodations.**

**Comment by dsjoerg** [comment]
@NanKrafft regarding your suggestion about the change to daily emails sent to Visitors, that sounds like it could be a new github issue separate from this one.  To help us get started on that I added a section to the end of the "Edits & Recommendations" Google Doc showing the current email copy that is sent to Visitors nightly if there are new potential hosts.

So, I think this issue is ready for implementation by a developer, limiting the changes to:
- When a visitor clicks "SEND MY CONTACT INFO", if their visit will begin within 48 hours, the email is sent immediately to the host (an "Instant Email").
- The (and for which we did not send an Instant Email) clause as per above
